Analysis/compare_dynasim_cps.qmd

```diff
@@ -0,0 +1,197 @@
+                ethncty %in% c(1000, 1999) ~ "White",
+                  TRUE ~ "Other"
+               )),
+           age_cat = as.factor(
+               case_when(
+                  age < 25 ~ "Under 25",
+                  age < 36 ~ "26-35",
+                  age < 46 ~ "36-45",
+                  age < 56 ~ "46-55",
+                  age < 66 ~ "56-65",
+                   TRUE ~ "Over 65"
+               )),
+            sex = as.factor(
+               case_when(
+                  sex == 1 ~ "Male",
+                  TRUE ~ "Female"
+               )))
+
+dyn_long$educ_clean <- factor(dyn_long$educ_clean, levels = c("0-11th grades", "HS degree", "Some college/No Degree", "Associate's Degree", "Bachelor's Degree", "More than Bachelor's"))
+
+dyn_long$age_cat <- factor(dyn_long$age_cat, levels = c("Under 25", "26-35", "36-45", "46-55", "56-65", "Over 65"))
+
+# Missings in CPS ASEC - about 24% excluded
+length(asec_df_long2$EDUC[asec_df_long2$EDUC==1])/nrow(asec_df_long2)
+length(asec_df_long2$EDUC[asec_df_long2$EDUC==999])/nrow(asec_df_long2)
+
+asec_df_long <- asec_df_long2 |>
+    filter(EDUC != 999 & EDUC != 1)|> #excluding missings, blanks or NIUs
+    mutate(
+        educ_clean = as.factor(
+               case_when(
+                  EDUC >= 0 & EDUC < 70 ~ "0-11th grades",
+                  EDUC  %in% c(70, 71, 72, 73) ~  "HS degree",  
+                  EDUC %in% c(80, 81) ~ "Some college/No Degree",
+                  EDUC %in% c(90, 91, 92, 100) ~ "Associate's Degree",
+                  EDUC %in% c(110,111) ~ "Bachelor's Degree",
+                  TRUE ~ "More than Bachelor's"
+               )),
+           race_eth = as.factor(
+              case_when(
+                  HISPAN != 0 ~ "Hispanic",
+                  RACE == 100 ~ "White",
+                  RACE == 200 ~ "Black",
+                  RACE == 651 ~ "Asian",
+                  TRUE ~ "Other"
+               )),
+            age_cat = as.factor(
+               case_when(
+                  AGE < 25 ~ "Under 25",
+                  AGE < 36 ~ "26-35",
+                  AGE < 46 ~ "36-45",
+                  AGE < 56 ~ "46-55",
+                  AGE < 66 ~ "56-65",
+                   TRUE ~ "Over 65"
+               )))
+
+asec_df_long$educ_clean <- factor(asec_df_long$educ_clean, levels = c("0-11th grades", "HS degree", "Some college/No Degree", "Associate's Degree", "Bachelor's Degree", "More than Bachelor's"))
+
+asec_df_long$age_cat <- factor(asec_df_long$age_cat, levels = c("Under 25", "26-35", "36-45", "46-55", "56-65", "Over 65"))
+
+asec_df_long <- janitor::clean_names(asec_df_long)
+
+plot_df = bind_rows(
+  select(asec_df_long, year, race_eth, age_cat, sex, educ_clean, asecwt) |> mutate(dataset='CPS ASEC'),
+  select(dyn_long, year, race_eth, age_cat, sex, educ_clean, asecwt) |> mutate(dataset='DYNASIM')) |>
+    filter(educ_clean != "0-11th grades" & educ_clean != "More than Bachelor's")
+
+
+```
+
+### Plots
+```{r}
+# Sex
+plot_df |> 
```

Comment:
These plots are good, but using facets makes them too small. Instead, I would like to try using tabsets (https://quarto.org/docs/interactive/layout.html#tabset-panel). In this case, we could have one tab for each education category and maybe use facets for gender. 

The simplest way is to create tabsets manually, but this is a lot of work. The other way is a little tricky, but it's much less work.

First, we would want to create plot_df as it enters the ggplot call, which means the count and mutate calls would modify plot_df. 

Then, we would create a chunk with option `asis` and surround it by panel tabset tags. The map call calls the function `plot_educ_by_educ` for each education level as defined in the above vector. (The backslash in the line with "```" is there only to have this show correctly.)
```
::: panel-tabset
```{r}
res = purrr::map_chr(educ_cats, plot_educ_by_educ)
cat(res, sep='\n')
\```
:::
```
The function contains your code for plotting but it returns it as a string. So when the map call above is executed, it returns the plotting code for each education level. And because of the "asis" options, Quarto interprets it as code that needs to be executed and not as text that needs to be shown. 
```
plot_educ_by_educ(eductab)
{
  knitr::knit_child(text = c(
    "#### `r eductab`",
    "",
    "```{r}",
    "#| echo: false",
    "plot_df %>%",
    "  filter(educ_clean==eductab) %>%",
    "  ggplot(aes(x=year, y=`Relative Freq`, color=dataset)) + geom_line() +",
    "  theme(legend.position = 'bottom')",
    "```",
    ""
  ), envir = environment(), quiet = TRUE)

}
```

Reply:
Hi Damir, thank you for sharing this idea! I really enjoyed learning about `tabset` and `asis`. I ended up also changing sex, because the years on the x-axis were being squished with `facet_grid`. Let me know what you think!